File: H-Index/I/H-IndexI.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        vector<int> indexs(citations.size() + 1, 0);
        int n = citations.size();
        for(auto c : citations) {
        	if(c > n)
        		indexs[n]++;
        	else
        		indexs[c]++;
        }
        int sum = n;
        for(int i = 0; i < indexs.size(); i++) {
        	if(sum - indexs[i] <= i)
        		return i;
        	sum -= indexs[i];
        }
        return 0;
    }
};

File: H-Index/I/H-IndexI.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        index = [0 for _ in range(len(citations) + 1)]
        N = len(citations)
        for c in citations:
            if c < N:
                index[c] += 1
            else:
                index[N] += 1
        sum = N
        for i in range(len(index)):
            if sum - index[i] <= i:
                return i
            sum -= index[i]
        return 0